l10n_ch_base_bank/models/invoice.py

```diff
@@ -11,6 +11,37 @@ class AccountInvoice(models.Model):
 
     _inherit = "account.invoice"
 
+    def _search(self, args, offset=0, limit=None, order=None, count=False,
+                access_rights_uid=None):
+        domain = []
+        for arg in args:
+            if not isinstance(arg, (tuple, list)) or len(arg) != 3:
+                domain.append(arg)
+                continue
+            field, operator, value = arg
+            if field != 'reference':
+                domain.append(arg)
+                continue
+            if operator not in ('like', 'ilike', '=like', '=ilike',
+                                'not like', 'not ilike', '=',
+                                '!=', '<', '<=', '>', '>=', 'in', 'not in'):
+                domain.append(arg)
+                continue
+            if value:
+                value = value.replace(' ', '')
```

Comment:
I think you need to handle the special case where value.replace(' ', '') -> empty string. This will match *all* invoices and kill your perfs, and it may happen with incremental searches.